src/stream/src/executor/managed_state/join/join_entry_state.rs

```diff
@@ -36,19 +37,34 @@ impl EstimateSize for JoinEntryState {
     }
 }
 
+#[derive(Error, Debug)]
+pub enum JoinEntryError {
+    #[error("double inserting a join state entry")]
+    OccupiedError,
+    #[error("removing a join state entry but it is not in the cache")]
+    RemoveError,
+}
+
```

Comment:
@BugenZhao do we need backtrace here?

Reply:
Since we don't panic in-place where the error occurs, I believe it's necessary. Additionally, this is an exceptional case, so the cost of capturing backtraces should not be a concern.

May use `thiserror_ext::Box` for this. https://docs.rs/thiserror-ext/latest/thiserror_ext/derive.Box.html#backtrace

Reply:
Makes sense. Backtraces captured in `join/mod.rs` should be sufficient.